tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/wizard/template/CDIWebProjectWizardTemplate.java

```diff
@@ -92,8 +97,13 @@ public void createCDIProject(){
 	
 	//cdi1.1+
 	//cd1.0 should show warning
+	//JBIDE-26643 | cdi2.0 does not contain the page for unchecking the "create beans.xml" option - test skiped
 	@Test
 	public void createCDIProjectWithoutBeansXml(){
+		// JBIDE-26643 | test skipped for CDI 2.0
+		if (CDIVersion.equals("2.0")) {
```

Comment:
I do not like the way how this is solved, imagine, when the problem is fixed, you have not other way to find out it is fixed unless you watch specific issue and you cannot catch any other possible failure if there is other bug. Rather, I would suggest to create a custom exception that will be thrown at the moment when specific issue occurs. So surround the erroneous code with try catch block, wait for proper exception, and throw your own exception (ie., JBIDEXYZException). In annotation of the test method add a expect param and assign it to your new exception. This design will allow you to have test green, unless there is another bug or issue is fixed.

Reply:
Changed as described. Now the test is running as expected for CDI 2.0, but not for CDI 1.0 and CDI 1.2, where no exception will be thrown (and the test expects the exception).

Reply:
Fixed - now the test is running for CDI 1.0, 1.2 a 2.0.